src/main/java/org/craftercms/studio/impl/v2/service/publish/PublishServiceImpl.java

```diff
@@ -230,6 +229,28 @@ public List<DeploymentHistoryGroup> getDeploymentHistory(String siteId, int days
         return groups;
     }
 
+    @Override
+    @HasPermission(type = CompositePermission.class, action = PERMISSION_PUBLISH)
+    public void publishAll(@ProtectedResourceId(SITE_ID_RESOURCE_ID) String siteId, String publishingTarget)
+            throws ServiceLayerException, UserNotFoundException {
+        publishServiceInternal.publishAll(siteId, publishingTarget);
+
+        SiteFeed siteFeed = siteService.getSite(siteId);
```

Comment:
Do we ignore audit records in case of an exception? If not this should be part of a finally

Reply:
as far as I know yeah, audit only records successful operations